aptos-move/block-executor/src/errors.rs

```diff
@@ -13,28 +14,29 @@ pub enum IntentionalFallbackToSequential {
     /// TODO: (short-mid term) relax the limitation, and (mid-long term) provide proper multi-versioning
     /// for code (like data) for the cache.
     ModulePathReadWrite,
-    /// We defensively check certain resource group related invariant violations.
-    ResourceGroupError(String),
+    /// We defensively check resource group serialization error in the commit phase.
+    /// TODO: should trigger invariant violation in the transaction itself.
+    ResourceGroupSerializationError(String),
 }
 
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub enum Error<E> {
+pub enum BlockError<E> {
```

Comment:
Why not even more descriptive - BlockExecutionEror

Reply:
It's too verbose, I deliberated as well, but since I suppose very theoretically there could be other Block related errors, I'll change it.